### PR TITLE
refactor(validations): group special post mainnet validations in a si…

### PR DESCRIPTION
**Pull request by girazoki**
…ngle file

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
Actually, this validation can simply be removed because old superblocks are not validated.

**Comment by girazoki** [comment]
For the moment, but at some point it might be useful to have that. In any case, I think we aare trying to set the basis for those validations to be somewhere

File: node/src/actors/chain_manager/mod.rs [hunk 1]
```diff
@@ -1331,13 +1334,11 @@ impl ChainManager {
...
-
                         let ars_members =
                             // Before reaching the epoch activity_period + collateral_age the bootstrap committee signs the superblock
                             // collateral_age is measured in blocks instead of epochs, but this only means that the period in which
                             // the bootstrap committee signs is at least epoch activity_period + collateral_age
-                            if is_superblock_hack_period {
+                            if is_first_superblock_hack_period(superblock_index) {
                                 // Bootstrap committee
                                 let hack_bootstrapping_committee = [
                                     "wit1asdpcspwysf0hg5kgwvgsp2h6g65y5kg9gj5dz",
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:1344]: Can we move this list of addresses to the new post_mainnet_validations.rs file

**Comment by aesedepece** [comment]
Sorry for nitpicking, but I wouldn't say those are *post-mainnet* validations but actually *mainnet* validations :stuck_out_tongue_winking_eye:

File: node/src/actors/chain_manager/mod.rs (hunk 1 above)
Comment (girazoki) [reply node/src/actors/chain_manager/mod.rs]: done

**Comment by aesedepece** [comment]
So funny that this mentions a "first" hack period hahaha

**Comment by girazoki** [comment]
I was already thinking about rebasing with the the second hack period that we prepared just in case hahaha

File: validations/src/mainnet_validations.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "wit1asdpcspwysf0hg5kgwvgsp2h6g65y5kg9gj5dz",
+    "wit13l337znc5yuualnxfg9s2hu9txylntq5pyazty",
+    "wit17nnjuxmfuu92l6rxhque2qc3u2kvmx2fske4l9",
+    "wit1drcpu0xc2akfcqn8r69vw70pj8fzjhjypdcfsq",
+    "wit1cyrlc64hyu0rux7hclmg9rxwxpa0v9pevyaj2c",
+    "wit1g0rkajsgwqux9rnmkfca5tz6djg0f87x7ms5qx",
+    "wit1etherz02v4fvqty6jhdawefd0pl33qtevy7s4z",
+];
+
+/// Returns the committee if the epoch provided corresponds to a hack period.
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:15]:
```suggestion
/// Return a hard-coded signing committee if the provided epoch belongs to an emergency period.
```

File: validations/src/mainnet_validations.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+use witnet_data_structures::chain::PublicKeyHash;
+use witnet_data_structures::chain::{Environment, Epoch};
+
+/// Committee for superblock indices 750-1344
+const FIRST_HACK_BOOTSTRAPPING_COMMITTEE: [&str; 7] = [
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:5]:
```suggestion
const FIRST_EMERGENCY_COMMITTEE: [&str; 7] = [
```

File: validations/src/mainnet_validations.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "wit17nnjuxmfuu92l6rxhque2qc3u2kvmx2fske4l9",
+    "wit1drcpu0xc2akfcqn8r69vw70pj8fzjhjypdcfsq",
+    "wit1cyrlc64hyu0rux7hclmg9rxwxpa0v9pevyaj2c",
+    "wit1g0rkajsgwqux9rnmkfca5tz6djg0f87x7ms5qx",
+    "wit1etherz02v4fvqty6jhdawefd0pl33qtevy7s4z",
+];
+
+/// Returns the committee if the epoch provided corresponds to a hack period.
+/// 750 and 1344: Between those indices, a special committee of 7 nodes was set.
+pub fn is_superblock_hack_period(
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:17]:
```suggestion
pub fn in_emergency_period(
```

File: validations/src/mainnet_validations.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_is_superblock_hack_period_not_mainnet() {
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:38]:
```suggestion
    fn test_in_emergency_period_not_mainnet() {
```

File: validations/src/mainnet_validations.rs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        assert_eq!(is_superblock_hack_period(1300, Environment::Testnet), None)
+    }
+    #[test]
+    fn test_is_superblock_hack_period_not_inside_period() {
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:42]:
```suggestion
    fn test_in_emergency_period_not_inside_period() {
```

File: validations/src/mainnet_validations.rs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        assert_eq!(is_superblock_hack_period(50, Environment::Mainnet), None)
+    }
+    #[test]
+    fn test_is_superblock_hack_period_inside_first_hack_period() {
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:46]:
```suggestion
    fn test_in_emergency_period_inside_first_emergency_period() {
```

File: validations/src/mainnet_validations.rs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        assert_eq!(
+            is_superblock_hack_period(800, Environment::Mainnet),
+            Some(
+                FIRST_HACK_BOOTSTRAPPING_COMMITTEE
+                    .iter()
+                    .map(|add| add.parse().expect("Malformed bootstrapping committee"))
+                    .collect(),
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:53]:
```suggestion
                FIRST_EMERGENCY_COMMITTEE
                    .iter()
                    .map(|address| address.parse().expect("Malformed signing committee"))
                    .collect(),
```

File: validations/src/mainnet_validations.rs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+pub fn is_superblock_hack_period(
+    superblock_index: Epoch,
+    environment: Environment,
+) -> Option<Vec<PublicKeyHash>> {
+    if Environment::Mainnet == environment && superblock_index > 750 && superblock_index < 1344 {
+        Some(
+            FIRST_HACK_BOOTSTRAPPING_COMMITTEE
+                .iter()
+                .map(|add| add.parse().expect("Malformed bootstrapping committee"))
+                .collect(),
```
Comment (aesedepece) [inline validations/src/mainnet_validations.rs:26]:
```suggestion
            FIRST_EMERGENCY_COMMITTEE
                .iter()
                .map(|add| add.parse().expect("Malformed signing committee"))
                .collect(),
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

**Review by aesedepece — APPROVED** [review]

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**